sources/packages/backend/apps/workers/src/services/student-assessment/student-assessment.service.ts

```diff
@@ -321,4 +332,108 @@ export class StudentAssessmentService extends RecordDataModelService<StudentAsse
       });
     }
   }
+
+  /**
+   * Get all outstanding student assessments to be calculated
+   * in the order of original assessment study start date.
+   * @param studentId student id.
+   * @param programYearId program year id.
+   * @returns student assessment to be calculated.
+   */
+  async getOutstandingAssessmentsForStudentInSequence(
+    studentId: number,
+    programYearId: number,
+  ): Promise<StudentAssessmentDetail[]> {
+    const originalAssessmentStudyStartDateAlias =
+      "originalAssessmentStudyStartDate";
+    // Sub query to get the original assessment study start date of a given assessment's application.
+    const originalAssessmentDateSubQuery = this.repo
```

Comment:
Even knowing that from a business perspective, we should have a maximum of one record, would it be worth adding a `limit 1` to the query?

Reply:
I agree.

Reply:
For my understanding. This change we are planning to limit to 1 as for a given application there will be only one original assessment right?

Reply:
It is limited to one because it needs only the first assessment as stated by the comment now: "Get first outstanding student assessment to be calculated".